src/tracing/step.ts

```diff
@@ -47,8 +47,8 @@ export async function traceWorkflowRunStep({
         "github.job.step.number": step.number,
         "github.job.step.started_at": step.started_at || undefined,
         "github.job.step.completed_at": step.completed_at || undefined,
+        "github.job.step.id": step.id,
         error: step.conclusion === "failure",
-        ...(step.id && { "github.job.step.id": step.id }),
```

Comment:
Was this breaking?

I figured that since not all steps have an ID, we should try to omit the trace span field in those cases where we don't find an ID on the step.